charmcraft/linters.py

```diff
@@ -523,6 +524,99 @@ def run(self, basedir: pathlib.Path) -> str:
         return self.Result.OK
 
 
+class OpsMainCall(Linter):
+    """Check that the entrypoint contains call to ops.main()."""
+
+    name = "ops-main-call"
+    url = f"{BASE_DOCS_URL}#heading--ops-main-call"
+    text = ""
+
+    def run(self, basedir: pathlib.Path) -> str:
```

Comment:
Could you add some tests for this method too please?

Reply:
What do you think, would unit tests that mock out everything be a good option?
Or a functional test that runs the linter against a zip file or unpacked source?

Reply:
An integration test would be ideal. You can make some fake minimal charm directories in `tests/integration/ops-main-linter-charms` and then parametrize with those and the expected outcomes.

Reply:
done, PTAL!